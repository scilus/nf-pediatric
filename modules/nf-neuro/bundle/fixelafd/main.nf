process BUNDLE_FIXELAFD {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
        tuple val(meta), path(bundles), path(fodf)

    output:
        tuple val(meta), path("*_afd_metric.nii.gz"), emit: fixel_afd
        path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    for bundle in $bundles;
        do\
        bname=\$(basename \${bundle} .trk)
        scil_compute_mean_fixel_afd_from_bundles.py \$bundle $fodf ${prefix}__\${bname}_afd_metric.nii.gz
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    scil_compute_mean_fixel_afd_from_bundles.py -h

    touch ${prefix}_test_afd_metric.nii.gz


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
