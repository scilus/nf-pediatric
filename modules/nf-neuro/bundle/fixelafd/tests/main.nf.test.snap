{
    "bundle - fixelafd": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__tracking_afd_metric.nii.gz:md5,01ac9796dc0ff6861c2992d513d690d3"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ad93266aca700e520e81c846f48d0ed1"
                ],
                "fixel_afd": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__tracking_afd_metric.nii.gz:md5,01ac9796dc0ff6861c2992d513d690d3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ad93266aca700e520e81c846f48d0ed1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-22T18:04:46.370916"
    }
}